### PR TITLE
Loosen nokogiri dependency

### PR DESCRIPTION
**Pull request by takiy33**
I loosen nokogiri dependency because i want to use nokogiri 1.7.0 or above.
Also, I add test target Rubies.

### Reference 
This PR is related #72.

### CODE REVIEW

File: test/jruby/html2haml_test.rb [hunk 1]
```diff
@@ -56,8 +56,7 @@ def test_style_to_css_filter_with_following_content
     foo {
         bar: baz;
     }
-%body
```
Comment (takiy33) [inline test/jruby/html2haml_test.rb]: I am not confident of this change. If you have idea, Please teach me 🙏

**Comment by benhutton** [comment]
:+1: 

Nokogiri 1.7.0 includes the removal of a very annoying deprecation warning: https://github.com/sparklemotion/nokogiri/commit/5b360b8bab3149c6ca706e6ac8321496ace2e6cb

For example, here's the output of my specs right now:

image

File: html2haml.gemspec [hunk 2]
```diff
@@ -17,7 +17,7 @@ Gem::Specification.new do |gem|
 
   gem.required_ruby_version = '>= 1.9.2'
 
-  gem.add_dependency 'nokogiri', '~> 1.6.0'
+  gem.add_dependency 'nokogiri', RUBY_VERSION < '2.1' ? '~> 1.6.0' : '>= 1.7'
```
Comment (benhutton) [inline html2haml.gemspec]: isn't it safer (and more generous) to use `'~> 1.6'`?

Comment (takiy33) [inline html2haml.gemspec]: Thank you for comment 👍 
When you `specify '~> 1.6'`, target version is `2.0 > version >= 1.6`.My intension is not to include Nokogiri 1.7 or above(Nokogiri required Ruby 2.1 or above since 1.7.0).

**Comment by amatsuda** [comment]
Firstly, switching dependency version in the gemspec does not work.
The gemspec in the repo will not be installed to the users' computers as it is.
The gemspec is used to generate the metadata in the gem package. So the nokogiri version statement will actually be evaluated when the gem owner is compiling a gem package. Not when the users are installing or loading the gem.

We can never control the users' runtime dependency gem versions. The only thing we can do here is to loosen the dependency version to accept all supported versions, and then specify "per ruby version" dependencies in Gemfile for the CI to pass.
That is why I showed you an example of switching the gem version in Gemfile in the [previous PR](https://github.com/haml/html2haml/pull/72#issuecomment-269443744).

**Comment by amatsuda** [comment]
Secondly, I'm not sure if the "Fix test" approach is right.
If only jruby + nokogiri 1.7 performs differently from other platforms, that can be considered a nokogiri or jruby bug.
In fact we're seeing exactly the same problem also in rails, https://travis-ci.org/rails/rails/jobs/188281629
and we're still not sure if it's a bug or nokogiri's intentional specification change.

**Comment by amatsuda** [comment]
Thirdly, please squash your four "Test against *" commits into one.

**Comment by takiy33** [comment]
Thank you for appreciate comment 😆 
I have do not enough time to do on weekdays, I will try on a holiday.

File: .travis.yml [hunk 3]
```diff
@@ -1,6 +1,9 @@
 language: ruby
 rvm:
-  - 2.1.0
+  - 2.4.0
+  - 2.3.3
+  - 2.2.6
+  - 2.1.10
```
Comment (oliverklee) [inline .travis.yml]: These changes are not related to the nokogiri depedency changes and should go into a separate PR.

**Comment by takiy33** [comment]
> Secondly, I'm not sure if the "Fix test" approach is right.
If only jruby + nokogiri 1.7 performs differently from other platforms, that can be considered a nokogiri or jruby bug.
In fact we're seeing exactly the same problem also in rails, https://travis-ci.org/rails/rails/jobs/188281629
and we're still not sure if it's a bug or nokogiri's intentional specification change.

Thank you for information. I understand my appoach is not good.
I will drop "Fix test" commit.

File: .travis.yml (hunk 3 above)
Comment (takiy33) [inline .travis.yml]: Your comment is good. But, now branching nokogiri version depending on Ruby version.
Please let me add this 🙏

Comment (oliverklee) [inline .travis.yml]: Ah, okay, makes sense.

Or even better: What do you think about adding the Ruby versions in a separate PR first? Then I'll create the PR.

Comment (oliverklee) [inline .travis.yml]: Sorry, I haven't had enough coffee yet. I already created the PR: #76 (also also #75 for dropping Ruby 1.9.3)

**Comment by amatsuda** [comment]
@takiy33 Thanks!
